src/test/scala/firrtlTests/ConstantPropagationTests.scala

```diff
@@ -908,6 +908,34 @@ class ConstantPropagationIntegrationSpec extends LowTransformSpec {
     execute(input, check, Seq(dontTouch("Child.in1")))
   }
 
+  it should "NOT optimize if no-constant-propagation is enabled" in {
+    val input =
+      """circuit Top :
+        |  module Child0 :
+        |    input in0 : UInt<1>
+        |    input in1 : UInt<1>
+        |    output out : UInt<1>
+        |    out <= and(in0, in1)
+        |  module Child1 :
+        |    input in0 : UInt<1>
+        |    input in1 : UInt<1>
+        |    output out : UInt<1>
+        |    out <= or(in0, in1)
+        |  module Top :
+        |    output z0 : UInt<1>
+        |    output z1 : UInt<1>
+        |    inst c0 of Child0
+        |    inst c1 of Child1
+        |    z0 <= c0.out
+        |    z1 <= c1.out
+        |    c0.in0 <= UInt<1>(1)
+        |    c0.in1 <= UInt<1>(1)
+        |    c1.in0 <= UInt<1>(0)
+        |    c1.in1 <= UInt<1>(0)""".stripMargin
+    val check = parse(input).serialize
+    execute(input, check, Seq(NoConstantPropagationAnnotation))
```

Comment:
This test case is pretty big for what it's checking. Can you trim this down to just one module? Something like the following:

```scala
circuit Foo:
  module Foo:
    input a: UInt<1>
    output b: UInt<1>
    b <= and(UInt<1>, a)
```

Reply:
Sure!